File: cpp-answers/119.杨辉三角-ii.cpp
/*
 * @Author: your name
 * @Date: 2020-06-05 20:35:37
 * @LastEditTime: 2020-06-05 20:45:17
 * @LastEditors: Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: \-Insgram-e:\leetcode刷题\119.杨辉三角-ii.cpp
 */
/*
 * @lc app=leetcode.cn id=119 lang=cpp
 *
 * [119] 杨辉三角 II
 */

// @lc code=start
class Solution
{
public:
    //参照118题的解法
    /*
    vector<int> getRow(int rowIndex)
    {
        vector<int> result;
        for (int i = 0; i <= rowIndex; i++)
        {
            result.push_back(1);
            for (int j = i - 1; j > 0; j--)
            {
                result[j] = result[j] + result[j - 1];
            }
        }
        return result;
    }*/
    //利用组合数公式
    vector<int> getRow(int rowIndex)
    {
        vector<int> result;
        result.push_back(1);
        long pre = 1;
        for (int k = 1; k <= rowIndex; k++)
        {
            long cur = pre * (rowIndex - k + 1) / k;
            result.push_back(cur);
            pre = cur;
        }
        return result;
    }
};
// @lc code=end

File: rust-answers/119.杨辉三角-ii.rs
/*
 * @lc app=leetcode.cn id=119 lang=rust
 *
 * [119] 杨辉三角 II
 */

// @lc code=start
impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let row_index = row_index as usize;
        let mut res = vec![];
        res.push(1);

        for i in 1..row_index + 1 {
            let tmp = res[i - 1] as usize * (row_index + 1 - i) / i;
            res.push(tmp as i32);
        }

        res
    }
}
// @lc code=end

